oauth2client/contrib/gce.py

```diff
@@ -48,28 +44,41 @@
 """
 
 
-def _get_service_account_email(http_request=None):
-    """Get the GCE service account email from the current environment.
-
-    Args:
-        http_request: callable, (Optional) a callable that matches the method
-                      signature of httplib2.Http.request, used to make
-                      the request to the metadata service.
-
-    Returns:
-        tuple, A pair where the first entry is an optional response (from a
-        failed request) and the second is service account email found (as
-        a string).
-    """
-    if http_request is None:
+def _get_metadata(http_request=None, *path):
+    if not http_request:
         http_request = httplib2.Http().request
+    full_path = "/".join(path.insert(0, _METADATA_ROOT))
     response, content = http_request(
-        _DEFAULT_EMAIL_METADATA, headers={'Metadata-Flavor': 'Google'})
+        full_path,
+        headers={'Metadata-Flavor': 'Google'}
+    )
     if response.status == http_client.OK:
-        content = _from_bytes(content)
-        return None, content
+        return json.loads(_from_bytes(content))
     else:
-        return response, content
+        raise AttributeError(
+            (
+                'Failed to retrieve {} from the Google Compute Engine'
+                'metadata service. Response:\n{}'
+            ).format(full_path, response)
```

Comment:
A lot going on here. Create the error message first and then just `raise AttributeError(msg)`